homeassistant/components/matter/adapter.py

```diff
@@ -74,13 +75,13 @@ def _setup_node(self, node: MatterNode) -> None:
             node.root_device_type_instance.get_cluster(all_clusters.BasicInformation)
         ):
             # create virtual (parent) device for bridge node device
-            bridge_device = MatterBridgedNodeDevice(
-                node.aggregator_device_type_instance
-            )
+            bridge_device = MatterNodeDevice(node)
```

Comment:
This is a wrong change. The class `MatterBridgedNodeDevice` exists for exactly this purpose here. It differentiates what data is being returned, as it's fundamentally different and you can't just combine them. 

For example, by removing the use of this type here, it breaks `_create_device_registry`

Reply:
it shouldn't really break `_create_device_registry` since that takes the `AbstractMatterNodeDevice` class which both `MatterBridgedNodeDevice` and `MatterNodeDevice` are subsets of, and my reasoning behind changing it is that the bridge itself is a matter device and it's just the devices it's bridging that are identified as bridged devices which are still identified as `MatterBridgedNodeDevice` types. with that being said I could just be miss interpreting how this is supposed to work and the main reason for this change is that `node.aggregator_device_type_instance` is of type `Aggregator` which doesn't have the `Clusters.BasicInformation` or the `Clusters.BridgedDeviceBasic` attribute so even if this were to be changed back to `MatterBridgedNodeDevice` it's data source would have to be changed to accommodate for this.

Reply:
It used to be typed to both the information cluster we found for root nodes and for bridged nodes. If that was changed, that might have to be revisited

Reply:
If that cluster doesn't exist anymore, we should first fix upstream, where we still define that those exist: https://github.com/home-assistant-libs/python-matter-server/blob/main/matter_server/common/models/node_device.py#L26

That property used to return device info that was aligned between both bridged devivec and other devices. If that does not work anymore, we should fix that upstream. It should return a method with basic information of the node, if we can't rely on overlapping property names between the two, we should introduce our own dataclass that holds information that we want.

Reply:
@ArturoGuerra thanks for your efforts so far! I am going to try another Hue bridge tomorrow so I can work with you to adjust the data model.